File: 프로그래머스/dfs_네트워크/네트워크.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;
vector<int> visited;
bool dfs(int x, vector<vector<int>>& comp){
    if(visited[x] == 1) return false;
    visited[x] = 1;
    for(int i =0 ; i < comp[x].size(); i++){
        if(visited[comp[x][i]] == 0 ){
            dfs(comp[x][i], comp);
        }
    }
     return true;
}
int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    vector<vector<int>> compo(201);
    for(int i=0; i < n; i++)
        visited.push_back(0);
    for(int i=0; i < computers.size(); i++){
        for(int j=0; j < computers[0].size(); j++){
            if (i!=j &&  computers[i][j] == 1) compo[i].push_back(j);
        }
    }
    /*
    for(int i=0;i < n; i++){
        for(int j= 0;j < compo[i].size(); j++)
            cout << compo[i][j] <<" ";
        cout << endl;
    }
    */
    for(int i=0; i < n; i++){
        if(dfs(i, compo))
            answer++;
    }
    return answer;
}

File: 프로그래머스/dfs_네트워크/네트워크.py
answer = 0
def dfs(comp,visited, idx ):
    global answer
    if visited[idx] == True:
        return False
    visited[idx] = True
    for i in comp[idx]:
        if not visited[i]:
            dfs(comp, visited, i)
    return True

def solution(n, computers):
    global answer
    visited = [False] * len(computers)
    graph = []
    
    for i in range(len(computers)):
        graph.append([])
        for j in range(len(computers)):
            if i!=j and computers[i][j] == 1 :
                graph[i].append(j)
    print(graph)
    for i  in range(len(computers)):
        if(dfs(graph, visited, i)):
            print(i)
            answer+=1
        
    return answer